src/Symfony/Component/Mailer/Bridge/Sendgrid/Transport/SendgridApiTransport.php

```diff
@@ -125,7 +126,13 @@ private function getPayload(Email $email, Envelope $envelope): array
                 continue;
             }
 
-            if ($header instanceof TagHeader) {
+            if ($name === 'send-at') {
+                if (!$header instanceof DateHeader) {
+                    throw new TransportException(sprintf('The Send-At header should be a "%s" instance to deduce the correct timestamp.', DateHeader::class));
+                }
+                // the documentation shows parameter names in camelCase but the expected JSON keys should be in snake_case
```

Comment:
```suggestion
```

There seems to be no point in explaining this for a specific case, since this rule applies to all parameters.

In addition, the documentation has an example of how to use:
https://www.twilio.com/docs/sendgrid/api-reference/mail-send/mail-send#request-body